### DISCUSSION TITLE
Documentation on which entities are queried directly from inverter and which are generated

### DISCUSSION THREAD

**Discussion by andrewmooreio** [nathanmarlor/foxess_modbus#347] [Support]
Hi,

Would it be possible to write up a wiki page differentiating between HA entity values which are pulled directly from the inverter, and which are generated using the integration?

It's good information to know, one will persist HA database wipes and the other won't.

As an example:

- sensor.load_power_total is generated by the integration using a reimann sum (I think the H1 does actually have this register, but it's not used)
- sensor.solar_sum_total is read directly from the inverter.

Thanks!

**Comment by canton7** [comment]
I've been wanting to auto-generate something like this for a bit. The complexity comes from the fact that we support 5 inverter models across 2 connection types, and eveything has a slightly different set of entities (and, indeed, an entity might be a Reimann sum for one combination of inverter / connection type, and read directly from the inverter for another).

I also want to implement entity schemas at some point (so we can start pulling load_power_total from the inverter without breaking people), which adds another dimension.

So any result is going to be fairly involved and hard to read, or involve a lot of duplication!

**Comment by andrewmooreio** [comment]
Makes sense, thanks. The fact that it varies between invert models definitely makes it tricky. Shall I raise it as a feature or leave it for now?
